### PR TITLE
[DO-1639] Pep single page

### PR DESCRIPTION
**Pull request by Iva-khar**

### CODE REVIEW

File: src/components/pages/datasets/DatasetsRoutes.jsx [hunk 1]
```diff
@@ -53,6 +54,11 @@ const DatasetsRoutes = (props) => {
         path={`${match.path}koatuu-fourth-level/`}
         component={KoatuuList}
       />
+      <Route
+        exact
+        path={`${match.path}pep/:idp`}
```
Comment (roman-tiukh) [inline src/components/pages/datasets/DatasetsRoutes.jsx:59]:
```suggestion
        path={`${match.path}pep/:id`}
```

File: src/components/pages/datasets/pep/PepDetail.jsx [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+import { ChevronDown, ChevronUp } from 'react-feather';
+import { useParams } from 'react-router-dom';
+import Api from 'api';
+
+
+const PepDetail = () => {
+  const { idp } = useParams();
+  const [data, setData] = useState([]);
+  const [error, setError] = useState('');
+  const [isDataReady, setDataReady] = useState(false);
```
Comment (roman-tiukh) [inline src/components/pages/datasets/pep/PepDetail.jsx:14]: isDataReady - useless state

File: src/components/pages/datasets/pep/PepDetail.jsx [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+import { useTranslation } from 'react-i18next';
+import { dateFormatISO } from 'utils';
+import { ChevronDown, ChevronUp } from 'react-feather';
+import { useParams } from 'react-router-dom';
+import Api from 'api';
+
+
+const PepDetail = () => {
+  const { idp } = useParams();
+  const [data, setData] = useState([]);
```
Comment (roman-tiukh) [inline src/components/pages/datasets/pep/PepDetail.jsx:12]: is `data` the array?

File: src/components/pages/datasets/pep/PepDetail.jsx [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+// import PropTypes from 'prop-types';
+import { useTranslation } from 'react-i18next';
+import { dateFormatISO } from 'utils';
+import { ChevronDown, ChevronUp } from 'react-feather';
+import { useParams } from 'react-router-dom';
+import Api from 'api';
+
+
+const PepDetail = () => {
+  const { idp } = useParams();
```
Comment (roman-tiukh) [inline src/components/pages/datasets/pep/PepDetail.jsx:11]:
```suggestion
  const { id } = useParams();
```

File: src/components/pages/datasets/pep/PepDetail.jsx [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+  const [data, setData] = useState([]);
+  const [error, setError] = useState('');
+  const [isDataReady, setDataReady] = useState(false);
+  const [isLoading, setLoading] = useState(false);
+  const { t, i18n } = useTranslation();
+  const fetchData = () => {
```
Comment (roman-tiukh) [inline src/components/pages/datasets/pep/PepDetail.jsx:17]:
```suggestion
  const { t, i18n } = useTranslation();
  
  const fetchData = () => {
```

File: src/components/pages/datasets/pep/PepDetail.jsx [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+          setError(err.response.data.detail);
+        }
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+  };
+  useEffect(() => {
+    fetchData();
+    console.log(data);
```
Comment (roman-tiukh) [inline src/components/pages/datasets/pep/PepDetail.jsx:35]: remove this `console.log(data);`

File: src/components/pages/datasets/pep/PepDetail.jsx [hunk 7]
```diff
@@ -0,0 +1,226 @@
...
+  }, []);
+  const fromPersonLinksLength = data.from_person_links ? data.from_person_links.length : 0;
+  const toPersonLinksLength = data.to_person_links ? data.to_person_links.length : 0;
```
Comment (roman-tiukh) [inline src/components/pages/datasets/pep/PepDetail.jsx:38]: ?

File: src/components/pages/datasets/pep/PepDetail.jsx [hunk 8]
```diff
@@ -0,0 +1,226 @@
...
+          <div className="pl-5 flex flex-row">
+            <div className="w-64 font-medium">{t('dateOfBirth')}:</div>
+            <div className="max-w-xl">{data.date_of_birth ? data.date_of_birth : '---'}</div>
+          </div>
+          <div className="pl-5 flex flex-row">
+            <div className="w-64 font-medium">{t('pepDetail.placeOfBirth')}:</div>
+            <div className="max-w-xl">{
+              (() => {
+                if (data.place_of_birth) {
+                  return i18n.language === 'uk' ? data.place_of_birth : data.place_of_birth_en;
```
Comment (roman-tiukh) [inline src/components/pages/datasets/pep/PepDetail.jsx:107]: just display both fields, and use ternary operators in jsx

File: src/components/pages/datasets/pep/PepDetail.jsx [hunk 9]
```diff
@@ -0,0 +1,226 @@
...
+          <div className="pl-5 flex flex-row">
+            <div className="w-64 font-medium">{t('pepDetail.updatedAt')}:</div>
+            <div className="max-w-xl">{dateFormatISO(data.updated_at)}</div>
+          </div>
+          <div className="pl-5 flex flex-row">
+            <div className="w-64 font-medium">{t('status')}:</div>
+            <div className="max-w-xl">{data.is_pep ? t('politicallyExposedPerson') : t('notPoliticallyExposedPerson')}</div>
+          </div>
+          <div className="pl-5 flex flex-row">
+            <div className="w-64 font-medium">{t('pepDetail.pepType')}:</div>
```
Comment (roman-tiukh) [inline src/components/pages/datasets/pep/PepDetail.jsx:87]: why pepDetail?

File: src/components/pages/datasets/pep/PepDetail.jsx [hunk 10]
```diff
@@ -0,0 +1,226 @@
...
+          <div className="pl-5 flex flex-row">
+            <div className="w-64 font-medium">{t('pepDetail.relatedPersons')}:</div>
+            <div className="max-w-xl">
+              <ul className={`overflow-y-hidden list-disc list-inside ${openRelatedPersons ? '' : 'max-h-3'}`}>
+                {fromPersonLinksLength ? data.from_person_links.map((person) => (
+                  <li>
+                    <span className="italic">{i18n.language === 'uk' ? person.to_person_relationship_type : person.to_person_relationship_type_en} — </span>
+                    <span className="capitalize underline">{i18n.language === 'uk' ? person.to_person.fullname : person.to_person.fullname_en}</span>
+                  </li>
+                )) : null}
```
Comment (roman-tiukh) [inline src/components/pages/datasets/pep/PepDetail.jsx:155]:
```suggestion
                {data.from_person_links.map((person) => (
                  <li>
                    <span className="italic">{i18n.language === 'uk' ? person.to_person_relationship_type : person.to_person_relationship_type_en} — </span>
                    <span className="capitalize underline">{i18n.language === 'uk' ? person.to_person.fullname : person.to_person.fullname_en}</span>
                  </li>
                ))}
```

File: src/components/pages/datasets/pep/PepDetail.jsx [hunk 11]
```diff
@@ -0,0 +1,226 @@
...
+            <div className="max-w-xl">{data.assets_info ? data.assets_info : '---'}</div>
+          </div>
+          <div className="pl-5 flex flex-row">
+            <div className="w-64 font-medium">{t('pepDetail.relatedPersons')}:</div>
+            <div className="max-w-xl">
+              <ul className={`overflow-y-hidden list-disc list-inside ${openRelatedPersons ? '' : 'max-h-3'}`}>
+                {fromPersonLinksLength ? data.from_person_links.map((person) => (
+                  <li>
+                    <span className="italic">{i18n.language === 'uk' ? person.to_person_relationship_type : person.to_person_relationship_type_en} — </span>
+                    <span className="capitalize underline">{i18n.language === 'uk' ? person.to_person.fullname : person.to_person.fullname_en}</span>
```
Comment (roman-tiukh) [inline src/components/pages/datasets/pep/PepDetail.jsx:153]: display both fields

File: src/components/pages/datasets/pep/PepDetail.jsx [hunk 12]
```diff
@@ -0,0 +1,226 @@
...
+                    <li>
+                      <span className="underline">
+                        {i18n.language === 'uk' ? company.company.name : company.company.name_en} ({company.company.edrpou})
+                      </span>
+                    </li>
+                  )) : '---'}
+              </ul>
+              {data.related_companies && data.related_companies.length > 3 ? (
+                <div className="flex flex-row cursor-pointer text-blue-800" onClick={() => toggleOpenRelatedCompanies()}>
+                  {openRelatedCompanies ? [<ChevronUp className="w-4 h-6" />, t('viewLess')] : [<ChevronDown className="w-4 h-6" />, t('viewMore')]}
```
Comment (roman-tiukh) [inline src/components/pages/datasets/pep/PepDetail.jsx:185]:
```suggestion
                  
                  {openRelatedCompanies ?  (
                    <>
                      <ChevronUp className="w-4 h-6" />
                      t('viewLess')
                    </>
                  ) : (
                    <>
                      <ChevronDown className="w-4 h-6" />
                      t('viewMore')
                    </>
                  )}
```

File: src/components/pages/datasets/pep/PepDetail.jsx [hunk 13]
```diff
@@ -0,0 +1,226 @@
...
+            <div className="max-w-xl">{data.criminal_record ? data.criminal_record : '---'}</div>
+          </div>
+          <div className="pl-5 flex flex-row">
+            <div className="w-64 font-medium">{t('pepDetail.criminalProceedings')}:</div>
+            <div className="max-w-xl">{
+              (() => {
+                if (data.criminal_proceedings) {
+                  const proceedings = data.criminal_proceedings.replace(/(\\r\\n)+/g, '</br>').replace(/(\\")+/g, '"');
+                  return openCriminalProceedings ?
+                    <div dangerouslySetInnerHTML={{ __html: proceedings }} /> :
```
Comment (roman-tiukh) [inline src/components/pages/datasets/pep/PepDetail.jsx:205]: really? you use `dangerouslySetInnerHTML`?

File: src/styles/pep-detail.scss [hunk 14]
```diff
@@ -0,0 +1,6 @@
+.max-h-3 {
+  max-height: 65px;
+}
+//.list-disc {
+//  list-style-type: disc;
+//}
```
Comment (roman-tiukh) [inline src/styles/pep-detail.scss:6]: move this code to utils.scss

**Review by roman-tiukh — CHANGES_REQUESTED** [review]

File: src/components/pages/datasets/DatasetsRoutes.jsx (hunk 1 above)
Comment (Iva-khar) [reply src/components/pages/datasets/DatasetsRoutes.jsx:59]: done

File: src/components/pages/datasets/pep/PepDetail.jsx (hunk 2 above)
Comment (Iva-khar) [reply src/components/pages/datasets/pep/PepDetail.jsx:14]: deleted

File: src/components/pages/datasets/pep/PepDetail.jsx (hunk 3 above)
Comment (Iva-khar) [reply src/components/pages/datasets/pep/PepDetail.jsx:12]: ?..
It behaves as array...

File: src/components/pages/datasets/pep/PepDetail.jsx (hunk 4 above)
Comment (Iva-khar) [reply src/components/pages/datasets/pep/PepDetail.jsx:11]: done

File: src/components/pages/datasets/pep/PepDetail.jsx (hunk 5 above)
Comment (Iva-khar) [reply src/components/pages/datasets/pep/PepDetail.jsx:17]: done

File: src/components/pages/datasets/pep/PepDetail.jsx (hunk 6 above)
Comment (Iva-khar) [reply src/components/pages/datasets/pep/PepDetail.jsx:35]: done

File: src/components/pages/datasets/pep/PepDetail.jsx (hunk 7 above)
Comment (Iva-khar) [reply src/components/pages/datasets/pep/PepDetail.jsx:38]: I need it for separate mapping from & to related_person_links & for considering to output view_more button

File: src/components/pages/datasets/pep/PepDetail.jsx (hunk 12 above)
Comment (Iva-khar) [reply src/components/pages/datasets/pep/PepDetail.jsx:185]: done

File: src/styles/pep-detail.scss (hunk 14 above)
Comment (Iva-khar) [reply src/styles/pep-detail.scss]: done

File: src/components/pages/datasets/pep/PepDetail.jsx (hunk 11 above)
Comment (Iva-khar) [reply src/components/pages/datasets/pep/PepDetail.jsx:152]: considered to leave as it is

File: src/components/pages/datasets/pep/PepDetail.jsx (hunk 8 above)
Comment (Iva-khar) [reply src/components/pages/datasets/pep/PepDetail.jsx:106]: rewrited

File: src/components/pages/datasets/pep/PepDetail.jsx (hunk 9 above)
Comment (Iva-khar) [reply src/components/pages/datasets/pep/PepDetail.jsx:86]: refactored

File: src/components/pages/datasets/pep/PepDetail.jsx (hunk 10 above)
Comment (Iva-khar) [reply src/components/pages/datasets/pep/PepDetail.jsx:154]: done

File: src/components/pages/datasets/pep/PepDetail.jsx (hunk 13 above)
Comment (Iva-khar) [reply src/components/pages/datasets/pep/PepDetail.jsx:224]: it fast solution

File: src/components/pages/datasets/pep/PepDetail.jsx (hunk 2 above)
Comment (roman-tiukh) [reply src/components/pages/datasets/pep/PepDetail.jsx:14]: its still useless state

File: src/components/pages/datasets/pep/PepDetail.jsx [hunk 15]
```diff
@@ -0,0 +1,248 @@
...
+  };
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+  const [openRelatedPersons, setOpenRelatedPersons] = useState(false);
+  const toggleOpenRelatedPersons = () => {
+    setOpenRelatedPersons(!openRelatedPersons);
+  };
+  const [openRelatedCompanies, setOpenRelatedCompanies] = useState(false);
```
Comment (roman-tiukh) [inline src/components/pages/datasets/pep/PepDetail.jsx:42]: all states must be declared at top of component

File: src/components/pages/datasets/pep/PepDetail.jsx [hunk 16]
```diff
@@ -0,0 +1,248 @@
...
+  const toggleOpenRelatedCompanies = () => {
+    setOpenRelatedCompanies(!openRelatedCompanies);
+  };
```
Comment (roman-tiukh) [inline src/components/pages/datasets/pep/PepDetail.jsx:45]: this function can be inline in jsx

File: src/components/pages/datasets/pep/PepDetail.jsx [hunk 17]
```diff
@@ -0,0 +1,248 @@
...
+  const [openCriminalProceedings, setOpenCriminalProceedings] = useState(false);
+  const toggleOpenCriminalProceedings = () => {
+    setOpenCriminalProceedings(!openCriminalProceedings);
+  };
+  return (isDataReady ? (
```
Comment (roman-tiukh) [inline src/components/pages/datasets/pep/PepDetail.jsx:50]: very bad ternary operator usage

File: src/components/pages/datasets/pep/PepDetail.jsx [hunk 18]
```diff
@@ -0,0 +1,248 @@
...
+            <div className="w-64 font-medium">{t('reasonOfTermination')}:</div>
+            <div className="max-w-xl">{data.reason_of_termination || '---'}</div>
+          </div>
+          <div className="pl-5 flex flex-row">
+            <div className="w-64 font-medium">{t('dateOfBirth')}:</div>
+            <div className="max-w-xl">{data.date_of_birth || '---'}</div>
+          </div>
+          <div className="pl-5 flex flex-row">
+            <div className="w-64 font-medium">{t('placeOfBirth')}:</div>
+            <div className="max-w-xl">{i18n.language === 'uk' ? data.place_of_birth || '---' : data.place_of_birth_en || '---'}
```
Comment (roman-tiukh) [inline src/components/pages/datasets/pep/PepDetail.jsx:102]: create function for get field on uk or en locale

File: src/components/pages/datasets/pep/PepDetail.jsx [hunk 19]
```diff
@@ -0,0 +1,248 @@
...
+          <div className="pl-5 flex flex-row">
+            <div className="w-64 font-medium">{t('lastPosition')}:</div>
+            <div className="max-w-xl">{
+              (() => {
+                if (data.last_job_title) {
+                  return i18n.language === 'uk' ? data.last_job_title : data.last_job_title_en;
+                }
+                return '---';
+              })()
+            }
```
Comment (roman-tiukh) [inline src/components/pages/datasets/pep/PepDetail.jsx:118]: no javascript in jsx

File: src/components/pages/datasets/pep/PepDetail.jsx [hunk 20]
```diff
@@ -0,0 +1,248 @@
...
+          <div className="pl-5 flex flex-row">
+            <div className="w-64 font-medium">{t('lastPlaceOfWork')}:</div>
+            <div className="max-w-xl">{
+              (() => {
+                if (data.last_employer) {
+                  return i18n.language === 'uk' ? data.last_employer : data.last_employer_en;
+                }
+                return '---';
+              })()
+            }
```
Comment (roman-tiukh) [inline src/components/pages/datasets/pep/PepDetail.jsx:130]: no javascript in jsx

File: src/components/pages/datasets/pep/PepDetail.jsx [hunk 21]
```diff
@@ -0,0 +1,248 @@
...
+                    <>
+                      <ChevronUp className="w-4 h-6" />
+                      {t('viewLess')}
+                    </>
+                  ) : (
+                    <>
+                      <ChevronDown className="w-4 h-6" />
+                      {t('viewMore')}
+                    </>
+                  )}
```
Comment (roman-tiukh) [inline src/components/pages/datasets/pep/PepDetail.jsx:167]: create component for this

File: src/components/pages/datasets/pep/PepDetail.jsx [hunk 22]
```diff
@@ -0,0 +1,248 @@
...
+                </div>
+              ) : null}
```
Comment (roman-tiukh) [inline src/components/pages/datasets/pep/PepDetail.jsx:199]: useless `null`

File: src/components/pages/datasets/pep/PepDetail.jsx [hunk 23]
```diff
@@ -0,0 +1,248 @@
...
+            <div className="max-w-xl">
+              <ul className={`overflow-y-hidden list-disc list-inside ${openRelatedCompanies ? '' : 'max-h-3'}`}>
+                {data.related_companies.length > 0 ?
+                  data.related_companies.map((company) => (
+                    <li>
+                      <span className="underline">
+                        {i18n.language === 'uk' ? company.company.name : company.company.name_en} ({company.company.edrpou})
+                      </span>
+                    </li>
+                  )) : '---'}
```
Comment (roman-tiukh) [inline src/components/pages/datasets/pep/PepDetail.jsx:183]:
```suggestion
                {data.related_companies.length > 0 ? data.related_companies.map((company) => (
                    <li>
                      <span className="underline">
                        {i18n.language === 'uk' ? company.company.name : company.company.name_en} ({company.company.edrpou})
                      </span>
                    </li>
                  )) : '---'}
```

File: src/components/pages/datasets/pep/PepDetail.jsx [hunk 24]
```diff
@@ -0,0 +1,248 @@
...
+              {i18n.language === 'uk' ? data.fullname : data.fullname_en}
+            </h2>
+          </div>
+          <div className="pl-5 flex flex-row">
+            <div className="w-64 font-medium">ID:</div>
+            <div className="max-w-xl">{data.id}</div>
+          </div>
+          <div className="pl-5 flex flex-row">
+            <div className="w-64 font-medium">{t('updatedAt')}:</div>
+            <div className="max-w-xl">{dateFormatISO(data.updated_at)}</div>
```
Comment (roman-tiukh) [inline src/components/pages/datasets/pep/PepDetail.jsx:78]: not ISO use dateFormat

File: src/components/pages/datasets/pep/PepDetail.jsx [hunk 25]
```diff
@@ -0,0 +1,248 @@
...
+    Api.get(`pep/${id}/`)
+      .then((resp) => {
+        setData(resp.data);
+        setDataReady(true);
+      })
+      .catch((err) => {
+        if (err.response?.data?.detail) {
+          setError(err.response.data.detail);
+        }
+      })
```
Comment (roman-tiukh) [inline src/components/pages/datasets/pep/PepDetail.jsx:29]: what is error? if error you sould just pop tooltip with message

File: src/components/pages/datasets/pep/PepDetail.jsx [hunk 26]
```diff
@@ -0,0 +1,248 @@
...
+import { useParams } from 'react-router-dom';
+import Api from 'api';
+
+
+const PepDetail = () => {
+  const { id } = useParams();
+  const [data, setData] = useState({});
+  const [error, setError] = useState('');
+  const [isDataReady, setDataReady] = useState(false);
+  const [isLoading, setLoading] = useState(false);
```
Comment (roman-tiukh) [inline src/components/pages/datasets/pep/PepDetail.jsx:15]: useless state?

File: src/components/pages/datasets/DatasetsRoutes.jsx [hunk 27]
```diff
@@ -53,6 +54,11 @@ const DatasetsRoutes = (props) => {
         path={`${match.path}koatuu-fourth-level/`}
         component={KoatuuList}
       />
+      <Route
+        exact
+        path={`${match.path}pep/:id`}
```
Comment (roman-tiukh) [inline src/components/pages/datasets/DatasetsRoutes.jsx:59]:
```suggestion
        path={`${match.path}pep/:id/`}
```

File: src/components/pages/datasets/pep/PepDetail.jsx [hunk 28]
```diff
@@ -0,0 +1,248 @@
...
+          </div>
+          <div className="pl-5 flex flex-row">
+            <div className="w-64 font-medium">{t('criminalProceedings')}:</div>
+            <div className="max-w-xl">{
+              (() => {
+                if (data.criminal_proceedings) {
+                  const proceedings = data.criminal_proceedings.replace(/(\\r\\n)+/g, '</br>').replace(/(\\")+/g, '"');
+                  return openCriminalProceedings ?
+                    <div dangerouslySetInnerHTML={{ __html: proceedings }} /> :
+                    <div dangerouslySetInnerHTML={{ __html: proceedings.substr(0, 777) }} />;
```
Comment (roman-tiukh) [inline src/components/pages/datasets/pep/PepDetail.jsx:218]: what is 777? I think its the wrong way

File: src/components/pages/datasets/pep/PepDetail.jsx (hunk 13 above)
Comment (roman-tiukh) [reply src/components/pages/datasets/pep/PepDetail.jsx:217]: no javascript in jsx

File: src/components/pages/datasets/pep/PepDetail.jsx [hunk 29]
```diff
@@ -0,0 +1,248 @@
...
+                }
+                return '---';
+              })()
+            }
+              {data.criminal_proceedings && data.criminal_proceedings.length > 777 ? (
```
Comment (roman-tiukh) [inline src/components/pages/datasets/pep/PepDetail.jsx:223]: 777....

**Review by roman-tiukh — CHANGES_REQUESTED** [review]

File: src/components/pages/datasets/pep/PepDetail.jsx (hunk 2 above)
Comment (Iva-khar) [reply src/components/pages/datasets/pep/PepDetail.jsx:14]: deleted

File: src/components/pages/datasets/pep/PepDetail.jsx (hunk 13 above)
Comment (Iva-khar) [reply src/components/pages/datasets/pep/PepDetail.jsx:217]: done

File: src/components/pages/datasets/pep/PepDetail.jsx (hunk 15 above)
Comment (Iva-khar) [reply src/components/pages/datasets/pep/PepDetail.jsx:42]: done

File: src/components/pages/datasets/pep/PepDetail.jsx (hunk 16 above)
Comment (Iva-khar) [reply src/components/pages/datasets/pep/PepDetail.jsx:45]: done

File: src/components/pages/datasets/pep/PepDetail.jsx (hunk 17 above)
Comment (Iva-khar) [reply src/components/pages/datasets/pep/PepDetail.jsx:50]: rewrited

File: src/components/pages/datasets/pep/PepDetail.jsx (hunk 18 above)
Comment (Iva-khar) [reply src/components/pages/datasets/pep/PepDetail.jsx:102]: done

File: src/components/pages/datasets/pep/PepDetail.jsx (hunk 19 above)
Comment (Iva-khar) [reply src/components/pages/datasets/pep/PepDetail.jsx:118]: done

File: src/components/pages/datasets/pep/PepDetail.jsx (hunk 20 above)
Comment (Iva-khar) [reply src/components/pages/datasets/pep/PepDetail.jsx:130]: done

File: src/components/pages/datasets/pep/PepDetail.jsx (hunk 21 above)
Comment (Iva-khar) [reply src/components/pages/datasets/pep/PepDetail.jsx:167]: done

File: src/components/pages/datasets/pep/PepDetail.jsx (hunk 22 above)
Comment (Iva-khar) [reply src/components/pages/datasets/pep/PepDetail.jsx:199]: rewrited

File: src/components/pages/datasets/pep/PepDetail.jsx (hunk 23 above)
Comment (Iva-khar) [reply src/components/pages/datasets/pep/PepDetail.jsx:183]: rewrited

File: src/components/pages/datasets/pep/PepDetail.jsx (hunk 24 above)
Comment (Iva-khar) [reply src/components/pages/datasets/pep/PepDetail.jsx:78]: done

File: src/components/pages/datasets/pep/PepDetail.jsx (hunk 25 above)
Comment (Iva-khar) [reply src/components/pages/datasets/pep/PepDetail.jsx:29]: rewrited

File: src/components/pages/datasets/pep/PepDetail.jsx (hunk 26 above)
Comment (Iva-khar) [reply src/components/pages/datasets/pep/PepDetail.jsx:15]: deleted

File: src/components/pages/datasets/DatasetsRoutes.jsx (hunk 27 above)
Comment (Iva-khar) [reply src/components/pages/datasets/DatasetsRoutes.jsx:59]: done

File: src/components/pages/datasets/pep/PepDetail.jsx (hunk 28 above)
Comment (Iva-khar) [reply src/components/pages/datasets/pep/PepDetail.jsx:218]: rewrited

**Comment by Iva-khar** [comment]
https://github.com/3v-workspace/DataOcean/pull/146#discussion_r659544071 - deleted;
https://github.com/3v-workspace/DataOcean/pull/146#discussion_r659551638 - done.

File: src/components/UnfoldingBlock.jsx [hunk 30]
```diff
@@ -0,0 +1,44 @@
...
+
+const UnfoldingBlock = (props) => {
+  const [isOpen, setOpen] = useState(false);
+  const [isOverflow, setOverflow] = useState(false);
+  const { children } = props;
+  const { t } = useTranslation();
+  const unfoldingWindowRef = useRef(null);
+
+  useEffect(() => {
+    if (unfoldingWindowRef.current) {
```
Comment (roman-tiukh) [inline src/components/UnfoldingBlock.jsx:13]: this `if` is needed?

File: src/components/UnfoldingBlock.jsx [hunk 31]
```diff
@@ -0,0 +1,44 @@
...
+      setOverflow(
+        unfoldingWindowRef.current.scrollHeight > unfoldingWindowRef.current.offsetHeight,
+      );
```
Comment (roman-tiukh) [inline src/components/UnfoldingBlock.jsx:16]: if scrollHeight > offsetHeight then setOverflow(true)

File: src/components/pages/datasets/pep/PepDetail.jsx [hunk 32]
```diff
@@ -0,0 +1,207 @@
...
+        {data.related_companies.map((company) => (
+          <li>
+            <span className="underline">
+              {getLocaleField(company.company, 'name')} ({company.company.edrpou})
+            </span>
+          </li>
+        ))}
+      </ul>
+    );
+  } else { relatedCompanies = '---'; }
```
Comment (roman-tiukh) [inline src/components/pages/datasets/pep/PepDetail.jsx:58]:
```suggestion
  } else {
    relatedCompanies = '---';
  }
```

File: src/components/pages/datasets/pep/PepDetail.jsx [hunk 33]
```diff
@@ -0,0 +1,207 @@
...
+import Api from 'api';
+import UnfoldingBlock from 'components/UnfoldingBlock';
+
+
+const PepDetail = () => {
+  const [data, setData] = useState({});
+  const { id } = useParams();
+  const { t, i18n } = useTranslation();
+  let relatedCompanies = '';
+  let relatedPersons = '';
```
Comment (roman-tiukh) [inline src/components/pages/datasets/pep/PepDetail.jsx:14]:
```suggestion
  let relatedCompanies = null;
  let relatedPersons = null;
```

File: src/components/pages/datasets/pep/PepDetail.jsx [hunk 34]
```diff
@@ -0,0 +1,207 @@
...
+            <span className="underline">
+              {getLocaleField(company.company, 'name')} ({company.company.edrpou})
+            </span>
+          </li>
+        ))}
+      </ul>
+    );
+  } else { relatedCompanies = '---'; }
+
+  if (data.from_person_links.length + data.to_person_links.length > 0) {
```
Comment (roman-tiukh) [inline src/components/pages/datasets/pep/PepDetail.jsx:60]: only the condition can be moved to jsx

File: src/components/pages/datasets/pep/PepDetail.jsx [hunk 35]
```diff
@@ -0,0 +1,207 @@
...
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  if (!Object.keys(data).length) {
+    return null;
+  }
+
+  if (data.related_companies.length > 0) {
```
Comment (roman-tiukh) [inline src/components/pages/datasets/pep/PepDetail.jsx:46]: only the condition can be moved to jsx

**Review by roman-tiukh — CHANGES_REQUESTED** [review]

File: src/components/UnfoldingBlock.jsx (hunk 30 above)
Comment (Iva-khar) [reply src/components/UnfoldingBlock.jsx:13]: deleted

File: src/components/UnfoldingBlock.jsx (hunk 31 above)
Comment (Iva-khar) [reply src/components/UnfoldingBlock.jsx:16]: done

File: src/components/pages/datasets/pep/PepDetail.jsx (hunk 32 above)
Comment (Iva-khar) [reply src/components/pages/datasets/pep/PepDetail.jsx:58]: done

File: src/components/pages/datasets/pep/PepDetail.jsx (hunk 33 above)
Comment (Iva-khar) [reply src/components/pages/datasets/pep/PepDetail.jsx:14]: done

File: src/components/pages/datasets/pep/PepDetail.jsx (hunk 34 above)
Comment (Iva-khar) [reply src/components/pages/datasets/pep/PepDetail.jsx]: rewrited

File: src/components/pages/datasets/pep/PepDetail.jsx (hunk 35 above)
Comment (Iva-khar) [reply src/components/pages/datasets/pep/PepDetail.jsx]: rewrited

**Review by roman-tiukh — APPROVED** [review]

**Review by EmmaSamoilova — APPROVED** [review]

**Review by AlenaYanish — APPROVED** [review]

**Pull request merged**